sql/src/main/java/io/crate/analyze/Rewriter.java

```diff
@@ -227,7 +206,7 @@ public Symbol apply(@Nullable RelationColumn input) {
             // if the column was only added to the outerSpec outputs because of the whereClause
             // it's possible to not collect it as long is it isn't used somewhere else
             if (!mssOutputSymbols.contains(symbol) &&
-                !SymbolVisitors.any(Predicates.<Symbol>equalTo(input), joinCondition)) {
+                !SymbolVisitors.any(Predicates.equalTo(input), joinCondition)) {
```

Comment:
I think this is still needed for Java7 and since we need to backport this fix...